components/learn/modules/ROOT/pages/on-dapps.adoc

```diff
@@ -1,8 +1,160 @@
+
+This means that, once we have deployed a contract and loaded it in our running React dapp, the app will **automatically pick up any changes to the Solidity code and upgrade the contract** during development. The https://github.com/OpenZeppelin/solidity-loader[hot loader] will watch all your Solidity contracts, and whenever you recompile, it will automatically upgrade the contract on the local development network - keeping the same address and state, and just changing the code to the new version you wrote.
+
+NOTE: The hot-loader is implemented as at https://webpack.js.org/[webpack] plugin. It will not work if you are using a different bundler for your application.
+
+To set up the hot-loader, you need to install `@openzeppelin/solidity-loader` and `json-loader`, and add the following https://webpack.js.org/configuration/module/#rule[rule] to your webpack configuration:
+
+```js
+test: /\.sol$/,
+use: [
+  { loader: 'json-loader' },
+  {
+    loader: '@openzeppelin/solidity-loader',
+    options: {
+      network: 'development',
+      disabled: false
+    },
+  },
+],
+```
+
+NOTE: If you are using `create-react-app`, you may need to either eject the application to modify the webpack configuration, or use https://github.com/timarney/react-app-rewired[`react-app-rewired`].
+
+Then, just load the contract Solidity file from your client code, which will go through the hot-loader. Whenever you modify your `Box` contract source, the loader will automatically upgrade the instance to the latest code for you.
+
+```js
+// Load web3 context using @openzeppelin/network
+const { networkId, lib: web3 } = useWeb3Network('http://127.0.0.1:8545');
+
+// Load contract artifact
+const Box = require('../../contracts/Box.sol');
+
+// Load deployed instance
+const [box, setBox] = useState(undefined);
+if (!box && Box && Box.networks && networkId) {
+  const deployedNetwork = Box.networks[networkId.toString()];
+  if (deployedNetwork) {
+    setBox(new web3.eth.Contract(Box.abi, deployedNetwork.address));
+  }
+}
+```
+
+You can follow a step-by-step tutorial on enabling hot loader in a React web application at https://forum.openzeppelin.com/t/building-an-openzeppelin-dapp-with-solidity-hot-loader/1843/1[the forum].
+
+[[starter-kit]]
+== Kickstarting your Dapp
+
+As you have seen so far, setting up a dapp may involve a certain deal of boilerplate and configuration, such as creating an OpenZeppelin CLI project, initializing a new webpack client app, setting up network configuration, loading your contracts, etc.
+
+To kickstart this process, we have built the xref:starter-kits[OpenZeppelin Starter Kits]. Starter Kits are preconfigured dapps with several OpenZeppelin libraries, Infura connections, and https://github.com/ConsenSys/rimble-ui[Rimble UI components], ready to start developing right away.
+
+You can start a new project from a starter kit using the `oz unpack` CLI command:
```

Comment:
I just noticed `oz unpack` requires the working directory to be empty, which goes against our recommendation to install the CLI locally (i.e. `npx oz unpack` doesn't work).

Reply:
Good catch. I'm creating an issue to rethink this, and removing the `npx` from this doc.

Reply:
https://github.com/OpenZeppelin/openzeppelin-sdk/issues/1374